src/module/vue/progress-sheet.vue

```diff
@@ -124,27 +124,22 @@
 </style>
 
 <script setup lang="ts">
-import SheetHeader from './sheet-header.vue'
 import { computed, inject, provide } from 'vue'
 import { RANKS, RANK_INCREMENTS } from '../constants'
 import { $ItemKey } from './provisions'
-import DocumentImg from './components/document-img.vue'
-import DocumentName from './components/document-name.vue'
 import RankPips from './components/rank-pips/rank-pips.vue'
 import BtnFaicon from './components/buttons/btn-faicon.vue'
-import ProgressTrack from './components/progress/progress-track.vue'
+import Track from './components/progress/track.vue'
 import Clock from './components/clock.vue'
 import MceEditor from './components/mce-editor.vue'
 import { throttle } from 'lodash'
 import SheetHeaderBasic from './sheet-header-basic.vue'
+import ProgressTrack from './components/progress/progress-track.vue'
 
+const props = defineProps<{ item: any }>()
 const $item = inject($ItemKey)
 
-const props = defineProps<{ item: any }>()
-provide(
-  'item',
-  computed(() => props.item)
-)
+provide($ItemKey, props.item)
```

Comment:
Two things. First, you're passing an `any` into a thing that is promising that the output will be an `IronswornItem`. And I'm pretty sure the thing you're pumping in there is **not** an `IronswornItem`, since those take extra care to pass as props.

And second, `$ItemKey` is already being provided; look two lines above this one. If you can inject a thing, you don't need to provide it.

Reply:
the `any` dates to a change you made in 03bf0267ffe4c4ca84f45534ee2e10c60f932ef8. i can certainly try to type it, but i'm not 100% clear on its original intent.

Reply:
hmm -- now that i think about it... if i understand it right (which, i might not): progress-sheet wouldn't be able to inject much on account of being a top-level component, wouldn't it? like there's nothing above it to provide $item for it.

Reply:
Let's take a walk! When Foundry tries to open this sheet:

1. It finds the registered class, [`ProgressSheetV2`](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/src/module/item/progress/progresssheet-v2.ts), and instantiates it
2. That class inherits from [`VueItemSheet`](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/src/module/vue/vueitemsheet.ts#L8), so now control passes to that class
3. `VueItemSheet` knows that it's attaching to an item, so it [gets ready](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/src/module/vue/vueitemsheet.ts#L22) to `provide($ItemKey, item)` when we get back here later. `VueActorSheet` does a similar thing with a Foundry actor instance.
4. Back in `ProgressSheetV2`, we load up a [Handlebars template](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/system/templates/item/progressv2.hbs) and let Foundry render it
5. Then control passes back to `VueItemSheet`'s render method, which [hands off](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/src/module/vue/vueitemsheet.ts#L38) to `VueSheetRenderHelper`, which is configured with a component mapping and a hook to do the `provide` back in step 3
6. [The helper class](https://github.com/ben/foundry-ironsworn/blob/eabf1cb9f25cabc888e7568b0c027fee66567092/src/module/vue/vue-render-helper.ts#L32) manages the Vue app instance, mounts it to a DOM element, and when the Foundry data is updated and a rerender is called for, updates the Vue-wrapped reactive thing, which does the in-place updating that we all love.

Now that I type it all out, I realize it's kind of complicated, but this is the best way I've seen to take the various desires of Vue and Foundry and make them work together. The Foundry/Vue seam is encapsulated in the helper class because Actor sheets have to be separate from Item sheets, and in the future will have to be separate from JournalEntryPage sheets.

Does it make sense?